lib/spack/spack/environment/environment.py

```diff
@@ -578,7 +603,72 @@ def specs_for_view(self, concretized_root_specs):
 
         return specs
 
-    def regenerate(self, concretized_root_specs):
+    def update_method_error_msg(self, methods):
+        """When methods are already determined invalid, construct error message for methods."""
+        msg = "View cannot be updated using specified update methods:"
+        if "exchange" in methods:
+            if not spack.util.atomic_update.renameat2():
+                msg += "\n  Operating system does not support 'exchange' atomic update method."
+                msg += f"\n    If the view {self.root} does not already exist on the filesystem,"
+                msg += " change its update_method to 'symlink' or 'auto'."
+                msg += f"\n    If the view at {self.root} exists already, either change the"
+                msg += " update_method and run `spack env view regenerate --force`"
+                msg += " or run on a newer OS."
```

Comment:
Can this helpful hint be removed?

Reply:
Can this "or run on a newer OS." comment still be removed? It's a weird suggestion